Sources/SwiftGRPCNIO/CallHandlers/BidirectionalStreamingCallHandler.swift

```diff
@@ -0,0 +1,44 @@
+import Foundation
+import SwiftProtobuf
+import NIO
+import NIOHTTP1
+
+/// Handles bidirectional streaming calls. Forwards incoming messages and end-of-stream events to the observer block.
+///
+/// - The observer block is implemented by the framework user and calls `context.sendResponse` as needed.
+/// - To close the call and send the status, fulfill `context.statusPromise`.
+public class BidirectionalStreamingCallHandler<RequestMessage: Message, ResponseMessage: Message>: BaseCallHandler<RequestMessage, ResponseMessage> {
+  public typealias EventObserver = (StreamEvent<RequestMessage>) -> Void
+  private var eventObserver: EventLoopFuture<EventObserver>?
+  
+  private var context: StreamingResponseCallContext<ResponseMessage>?
```

Comment:
Can we remove this? Looks like it should be captured by the closure below when it's used

Reply:
No; the context should live for as long as the handler does. The closure, however, has a much shorter lifetime.